src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue

```diff
@@ -0,0 +1,236 @@
+              display-type="secondary"
+              :label="$t('performance.activate')"
+              type="button"
+              @click="onActivate"
+            />
+          </div>
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@/core/util/helper/navigation';
+import EmployeeAutocomplete from '@/core/components/inputs/EmployeeAutocomplete';
+import SupervisorAutoComplete from '@/orangehrmPerformancePlugin/components/SupervisorAutoComplete';
+import {APIService} from '@/core/util/services/api.service';
+import {
+  endDateShouldBeAfterStartDate,
+  required,
+  startDateShouldBeBeforeEndDate,
+  validDateFormat,
+} from '@/core/util/validation/rules';
+
+const reviewModel = {
+  employee: null,
+  supervisorReviewer: null,
+  startDate: null,
+  endDate: null,
+  dueDate: null,
+};
+
+export default {
+  components: {
+    'employee-autocomplete': EmployeeAutocomplete,
+    'supervisor-autocomplete': SupervisorAutoComplete,
+  },
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/performance/reviews',
+    );
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      review: {...reviewModel},
+      rules: {
+        employee: [required],
+        supervisorReviewer: [required],
+        startDate: [
+          required,
+          validDateFormat(),
+          startDateShouldBeBeforeEndDate(
+            () => this.review.endDate,
+            this.$t(
+              'general.review_period_start_date_should_be_before_end_date',
+            ),
+            {allowSameDate: false},
+          ),
+        ],
+        endDate: [
+          required,
+          validDateFormat(),
+          endDateShouldBeAfterStartDate(
+            () => this.review.startDate,
+            this.$t(
+              'performance.review_period_end_date_should_be_after_start_date',
+            ),
+          ),
+        ],
+        dueDate: [
+          required,
+          validDateFormat(),
+          endDateShouldBeAfterStartDate(
+            () => this.review.endDate,
+            this.$t(
+              'performance.due_date_should_be_after_review_period_end_date',
+            ),
+          ),
+        ],
+      },
+    };
+  },
+  methods: {
+    onCancel() {
+      navigate('/performance/searchPerformanceReview');
+    },
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .create({
+          empNumber: this.review.employee.id,
+          reviewer: this.review.supervisorReviewer.id,
+          startDate: this.review.startDate,
+          endDate: this.review.endDate,
+          dueDate: this.review.dueDate,
+          activate: false,
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .then(() => {
+          // go back
+          this.onCancel();
+        });
+    },
+    onActivate() {
```

Comment:
what if we take both `onActivate()` and `onSave()` to one function?

```
onSave(activate = false) {
  this.isLoading = true;
  this.http
    .create({
      empNumber: this.review.employee.id,
      reviewer: this.review.supervisorReviewer.id,
      startDate: this.review.startDate,
      endDate: this.review.endDate,
      dueDate: this.review.dueDate,
      activate,
    })
    .then(() => {
      return this.$toast.saveSuccess();
    })
    .catch((error) => {
      return this.$toast.warn({
        title: this.$t("general.warning"),
        message: error,
      });
    })
    .then(() => {
      this.onCancel();
    });
}
```
finally in your activate button `@click="onSave(true)"` and for form submit `@submitValid="onSave(false)"`